common/app/common/JsonComponent.scala

```diff
@@ -12,6 +12,12 @@ import play.api.http.ContentTypes._
 
 object JsonComponent extends Results with implicits.Requests {
 
+  def withRefreshStatus(json: JsValue): JsValue =
+    json.as[JsObject] + ("refreshStatus" -> toJson(AutoRefreshSwitch.isSwitchedOn))
+
+  def apply[A](value: A)(implicit request: RequestHeader, writes:Writes[A]) =
+    resultFor(request, Json.stringify(withRefreshStatus(toJson(value))))
```

Comment:
Good catch! I'm changing this to now check to see whether `value` is of type `JsObject` and, if so, the `refreshStatus` will be added to it, otherwise it won't be. I believe that this then leaves all behaviour unchanged.